internal/aws/resource_aws_eks_availability.go

```diff
@@ -0,0 +1,318 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"connectrpc.com/connect"
+	config_client "github.com/common-fate/sdk/config"
+	configv1alpha1 "github.com/common-fate/sdk/gen/commonfate/control/config/v1alpha1"
+	entityv1alpha1 "github.com/common-fate/sdk/gen/commonfate/entity/v1alpha1"
+	"github.com/common-fate/sdk/service/control/configsvc"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type AWSEKSAvailability struct {
```

Comment:
This one should be the AWSEKSAvailabilities because it is configuring the selector which creates more than one availabilies 
```suggestion
type AWSEKSAvailabilities struct {
```